asynction/types.py

```diff
@@ -13,11 +15,223 @@
+        elif (
+            self.authorization_code is not None
+            and self.authorization_code.token_url is None
+        ):
+            raise ValueError("Authorization code OAuth flow is missing Token URL")
+
+    def supported_scopes(self) -> Iterator[str]:
+        # note Cannot lru_cache this
+        # TypeError: unhashable type: 'OAuth2Flows'
+        for f in fields(self):  # dataclasses.fields
+            flow = getattr(self, f.name)
+            if flow:
+                for scope in flow.scopes:
+                    yield scope
+
+
+register_forge(OAuth2Flows, OAuth2Flows.forge)
+
+
+class SecuritySchemesType(Enum):
+    """
+    https://www.asyncapi.com/docs/specifications/v2.2.0#securitySchemeObjectType
+    """
+
+    USER_PASSWORD = "userPassword"
+    API_KEY = "apiKey"
+    X509 = "X509"
+    SYMMETRIC_ENCRYPTION = "symmetricEncryption"
+    ASYMMETRIC_ENCRYPTION = "asymmetricEncryption"
+    HTTP_API_KEY = "httpApiKey"
+    HTTP = "http"
+    OAUTH2 = "oauth2"
+    OPENID_CONNECT = "openIdConnect"
+    PLAIN = "plain"
+    SCRAM_SHA256 = "scramSha256"
+    SCRAM_SHA512 = "scramSha512"
+    GSSAPI = "gssapi"
+
+
+@dataclass
+class SecurityScheme:
+    """
+    https://www.asyncapi.com/docs/specifications/v2.2.0#securitySchemeObject
+    """
+
+    type: SecuritySchemesType
+    description: Optional[str] = None
+    name: Optional[str] = None  # Required for httpApiKey
+    in_: Optional[str] = None  # Required for httpApiKey | apiKey
+    scheme: Optional[HTTPAuthenticationScheme] = None  # Required for http
+    bearer_format: Optional[str] = None  # Optional for http ("bearer")
+    flows: Optional[OAuth2Flows] = None  # Required for oauth2
+    open_id_connect_url: Optional[str] = None  # Required for openIdConnect
+
+    x_basic_info_func: Optional[str] = None  # Required for http(basic)
+    x_token_info_func: Optional[str] = None  # Required for oauth2
+    x_api_key_info_func: Optional[str] = None  # Required for apiKey
+    x_scope_validate_func: Optional[str] = None  # Optional for oauth2
+
+    def __post_init__(self):
+        if not self.flows and self.type in [
+            SecuritySchemesType.OAUTH2,
+            SecuritySchemesType.OPENID_CONNECT,
+        ]:
+            raise ValueError(
+                "flows field should be be defined " f"for {self.type} security schemes"
+            )
+
+        if self.type is SecuritySchemesType.HTTP:
+            # NOTE bearer_format is optional for HTTP
+            if not self.scheme:
+                raise ValueError(f"scheme is required for {self.type} security schemes")
+
+        if self.type is SecuritySchemesType.HTTP_API_KEY:
+            options = ["query", "header", "cookie"]
+            if not self.in_ or self.in_ not in options:
+                raise ValueError(
+                    f'"in" field must be one of {options} for {self.type} security schemes'  # noqa: 501
+                )
```

Comment:
```suggestion
                raise ValueError(
                    f'"in" field must be one of {options} '
                    f'for {self.type} security schemes'
                )
```
You can split string lines like this ☝️ 
Would avoid using the `noqa` comment unless it's necessary